tasks/brolly_stats_rollup.py

```diff
@@ -0,0 +1,132 @@
+import asyncio
+import datetime
+import json
+import logging
+import uuid  # TODO remove
+
+import httpx
+from shared.celery_config import brolly_stats_rollup_task_name
+from shared.config import get_config
+from shared.metrics import metrics
+
+from app import celery_app
+from database.models import Commit, Constants, Repository, Upload, User
+from tasks.crontasks import CodecovCronTask
+
+log = logging.getLogger(__name__)
+
+DEFAULT_BROLLY_ENDPOINT = "https://codecov.io/self-hosted/telemetry"
+
+
+class BrollyStatsRollupTask(CodecovCronTask):
+    """
+    By default, this cron task collects anonymous statistics about the Codecov instance
+    and submits them to Codecov to give us insight into the size of our self-hosted,
+    open-source userbase.
+
+    Installations can configure the behavior in `codecov.yml`:
+    - `setup.telemetry.enabled` (default True): Control whether stats are collected at all.
+    - `setup.telemetry.endpoint_override`: Control where stats are sent.
+    - `setup.telemetry.anonymous` (default True): Request that brolly not save identifiable
+      information such as the IP address of the installation that submitted the stats.
+    - `setup.telemetry.admin_email`: Contact information for the installation owner.
+    """
+
+    @classmethod
+    def get_min_seconds_interval_between_executions(cls):
+        return 72000  # 20h
+
+    name = brolly_stats_rollup_task_name
+
+    def _get_install_id(self, db_session):
+        """
+        Anonymous, randomly-generated UUID created during DB setup.
+        """
+        return db_session.query(Constants).get("install_id").value
+
+    def _get_users_count(self, db_session):
+        return db_session.query(User).count()
```

Comment:
This might be a silly question. It's valid for all the following models as well.

I _think_ that if you query the class `sqlAlchemy` will be eager and return everything. If you just need the count, would it not be more efficient to do something like `db_session.query(User.id).count()`?

Reply:
https://stackoverflow.com/questions/14754994/why-is-sqlalchemy-count-much-slower-than-the-raw-query

seems so, thanks for spotting! will update